addons/mrp/models/mrp_unbuild.py

```diff
@@ -202,7 +202,7 @@ def action_unbuild(self):
                 move.quantity = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
 
         (finished_moves | consume_moves | produce_moves).picked = True
-        finished_moves._action_done()
+        finished_moves.with_context(unbuild_mo_id=self.mo_id)._action_done()
```

Comment:
If the unbuild order comes from a manufacturing order, the moves are linked via `origin_returned_move_id`. See 

https://github.com/odoo/odoo/blob/a848c3854c94b5c2b752edddbcf49337acf9d6ea/addons/mrp/models/mrp_unbuild.py#L268

I think is more robust than a context and a search